internal/subscription/core.go

```diff
@@ -343,54 +342,39 @@ func (c *Core) Migrate(ctx context.Context, names []string) error {
-	for _, model := range subscriptionsToUpdate {
-		needsUpdate := false
-
-		// update retry policy if not set
-		if model.RetryPolicy == nil {
-			model.setDefaultRetryPolicy()
-			// update dead letter policy every time as previously we were saving only dead-letter topic names
-			// this takes care of updating max_delivery_attempts
-			model.setDefaultDeadLetterPolicy()
-			needsUpdate = true
+	updatedSubCount := 0
+	for _, sub := range subscriptionsToUpdate {
+		if sub.PushConfig == nil {
+			continue
 		}
-
-		if needsUpdate {
-			// collect all the delay topic names to be created
-			topicNames = append(topicNames, model.GetDelayTopics()...)
-
-			logger.Ctx(ctx).Infow("migration: updating subscription", "model", model.Name)
-			// update the subscription model
-			err := c.UpdateSubscription(ctx, model)
-			if err != nil {
-				return err
+		pushAttr := sub.PushConfig.Attributes
+		if pushAttr != nil {
+			needsUpdate := false
+			if _, ok := pushAttr[attributeUsername]; ok {
+				logger.Ctx(ctx).Infow("migration: deleting username", "username", pushAttr[attributeUsername], "subscription", sub.Name)
+				delete(pushAttr, attributeUsername)
+				needsUpdate = true
 			}
-		}
-	}
 
-	logger.Ctx(ctx).Infow("migration: delay topics to create", "count", len(topicNames), "topicNames", topicNames)
-	// collect the success and failed topic names so that they can be retried if needed
-	successTopicNames, failedTopicNames := make([]string, 0), make([]string, 0)
-	for _, tName := range topicNames {
-		tModel, terr := topic.GetValidatedModel(ctx, &metrov1.Topic{
-			Name: tName,
-		})
-		if terr != nil {
-			logger.Ctx(ctx).Errorw("migration: failed to create validated topic model", "tName", tName, "error", terr.Error())
-			failedTopicNames = append(failedTopicNames, tName)
-			continue
-		}
+			if _, ok := pushAttr[attributePassword]; ok {
+				logger.Ctx(ctx).Infow("migration: deleting password", "password", pushAttr[attributePassword], "subscription", sub.Name)
+				delete(pushAttr, attributePassword)
+				needsUpdate = true
+			}
 
-		err := c.topicCore.CreateTopic(ctx, tModel)
-		if err != nil {
-			logger.Ctx(ctx).Errorw("migration: failed to create delay topic", "tName", tName, "error", err.Error())
-			failedTopicNames = append(failedTopicNames, tName)
-			continue
+			if needsUpdate {
+				updatedSubCount++
+				logger.Ctx(ctx).Infow("migration: updating subscription", "subscription", sub.Name)
+				sub.PushConfig.Attributes = pushAttr
+				err := c.UpdateSubscription(ctx, sub)
+				if err != nil {
+					logger.Ctx(ctx).Errorw("migration: error in updating subscription", "subscription", sub.Name, "error", err.Error())
+					return err
+				}
+			}
 		}
-		successTopicNames = append(successTopicNames, tName)
 	}
 
-	logger.Ctx(ctx).Infow("migration: request completed", "successTopicNames", successTopicNames, "failedTopicNames", failedTopicNames)
+	logger.Ctx(ctx).Infow("migration: request completed.", "Subscriptions Updated", updatedSubCount)
```

Comment:
```suggestion
	logger.Ctx(ctx).Infow("migration: request completed.", "subscriptionsUpdated", updatedSubCount)
```
keep this as one word, easier to do an exact match on Sumo.